backend/src/supervisors/supervisor.py

```diff
@@ -16,16 +16,24 @@ async def solve_all(intent_json) -> None:
     questions = intent_json["questions"]
 
     if len(questions) == 0:
-        raise Exception(no_questions_response)
-
-    for question in questions:
         try:
+            question = intent_json["question"]
             (agent_name, answer, status) = await solve_task(question, get_scratchpad())
             update_scratchpad(agent_name, question, answer)
             if status == "error":
                 raise Exception(answer)
         except Exception as error:
             update_scratchpad(error=str(error))
+    else:
```

Comment:
It makes sense to have a process_question func and then call it in both if and else

async def process_question(question):
        try:
            agent_name, answer, status = await solve_task(question, get_scratchpad())
            update_scratchpad(agent_name, question, answer)
            if status == "error":
                logger.info("status == error")
                raise Exception(answer)
        except Exception as error:
            update_scratchpad(error=str(error))
            

if not questions:
        question = intent_json.get("question")
        if question:
            await process_question(question)
    else:
        for question in questions:
            await process_question(question)